LoopKit/DeviceManager/DeviceLog/PersistentDeviceLog.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  PersistentDeviceLog.swift
+//  LoopKit
+//
+//  Created by Pete Schwamb on 1/13/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import CoreData
+import os.log
+
+
+// Using a framework specific class will search the framework's bundle for model files.
+class PersistentContainer: NSPersistentContainer { }
+
+public class PersistentDeviceLog {
+    
+    private let storageFile: URL
+    
+    private let managedObjectContext: NSManagedObjectContext
+
+    private let persistentContainer: NSPersistentContainer
+    
+    private let maxAge: TimeInterval
```

Comment:
[nit] I took this to be the `maxAge` of the `PersistentDeviceLog`, which was confusing. But now I understand it is the `maxAge` of a log entry. Should it be updated to `maxEntryAge`? Alternatively, is the intent was to be the age of the log itself, should it be updated to `logDuration`?